packages/bigtable/src/instance.js

```diff
@@ -0,0 +1,665 @@
+     */
+    delete: {
+      protoOpts: {
+        service: 'BigtableInstanceAdmin',
+        method: 'deleteInstance'
+      },
+      reqOpts: {
+        name: id
+      }
+    },
+
+    /**
+     * Check if an instance exists.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {boolean} callback.exists - Whether the instance exists or not.
+     *
+     * @example
+     * instance.exists(function(err, exists) {});
+     */
+    exists: true,
+
+    /**
+     * Get an instance if it exists.
+     *
+     * @example
+     * instance.get(function(err, instance, apiResponse) {
+     *   // The `instance` data has been populated.
+     * });
+     */
+    get: true,
+
+    /**
+     * Get the instance metadata.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.metadata - The metadata.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * instance.getMetadata(function(err, metadata, apiResponse) {});
+     */
+    getMetadata: {
+      protoOpts: {
+        service: 'BigtableInstanceAdmin',
+        method: 'getInstance'
+      },
+      reqOpts: {
+        name: id
+      }
+    },
+
+    /**
+     * Set the instance metadata.
+     *
+     * @param {object} metadata - Metadata object.
+     * @param {string} metadata.displayName - The descriptive name for this
+     *     instance as it appears in UIs. Can be changed at any time, but should
+     *     be kept globally unique to avoid confusion.
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * instance.setMetadata({
+     *   displayName: 'updated-name'
+     * }, function(err, apiResponse) {});
+     */
+    setMetadata: {
+      protoOpts: {
+        service: 'BigtableInstanceAdmin',
+        method: 'updateInstance'
+      },
+      reqOpts: {
+        name: id
+      }
+    }
+  };
+
+  var config = {
+    parent: bigtable,
+    id: id,
+    methods: methods,
+    createMethod: function(_, options, callback) {
+      bigtable.createInstance(name, options, callback);
+    }
+  };
+
+  common.GrpcServiceObject.call(this, config);
+}
+
+nodeutil.inherits(Instance, common.GrpcServiceObject);
+
+/**
+ * Formats the full table name into a user friendly version.
+ *
+ * @private
+ *
+ * @param {string} name - The formatted Table name.
+ * @return {string}
+ *
+ * @example
+ * Instance.formatTableName_('projects/p/zones/z/clusters/c/tables/my-table');
+ * // => 'my-table'
+ */
+Instance.formatTableName_ = function(name) {
+  if (name.indexOf('/') === -1) {
```

Comment:
I know this is copied from other parts of our lib, but looking at it now, I'm pretty sure we can just do:

```js
return name.split('/').pop();
```